pkg/recipes/terraform/execute_test.go

```diff
@@ -114,3 +115,62 @@ func Test_GetTerraformConfig_InvalidDirectory(t *testing.T) {
 	require.Error(t, err)
 	require.Contains(t, err.Error(), "error creating file: open invalid-directory/main.tf.json: no such file or directory")
 }
+
+func TestSetEnvironmentVariables(t *testing.T) {
+	testCase := []struct {
+		name string
+		opts Options
+	}{
+		{
+			name: "set environment variables",
+			opts: Options{
+				EnvConfig: &recipes.Configuration{
+					RecipeConfig: dm.RecipeConfigProperties{
+						EnvVariables: dm.EnvironmentVariables{
+							AdditionalProperties: map[string]any{
+								"TEST_ENV_VAR1": "value1",
+								"TEST_ENV_VAR2": "value2",
```

Comment:
Can you please add coverage for scenarios where value is not a string and another one for nil value?

Reply:
updated types per design discussion to map[string]string. value now will always be string